lib/common/patchset.c

```diff
@@ -703,15 +700,12 @@ apply_v2_patchset(xmlNode *xml, const xmlNode *patchset)
             } else if (match->last) {
                 crm_trace("Adding %s at position %d (end)",
                           child->name, position);
-                xmlAddNextSibling(match->last, child);
 
             } else {
                 crm_trace("Adding %s at position %d (first)",
                           child->name, position);
                 CRM_LOG_ASSERT(position == 0);
-                xmlAddChild(match, child);
             }
```

Comment:
At this point we might as well consolidate the last two blocks and drop the log assert